### PR TITLE
add user sign up

### PR DESCRIPTION
**Pull request by petushka1**
# Scope
- [x] Can register a new user.
- [x] User logs in with a combination of email and password.
- [x] Hashed password should be stored in the database.
- [x] Ask for confirmation of email.
- [x] Can reset password.

### CODE REVIEW

File: config/routes.rb [hunk 1]
```diff
@@ -1,4 +1,8 @@
 Rails.application.routes.draw do
+  devise_for :users, path: '/', path_names: {
+    sign_in: 'login'
+  }
```
Comment (julie-ify) [inline config/routes.rb:4]: - [ ] After signing in I get this error. We can fix it by changing to this:

  ```
  devise_for :users
  ```

  kru

File: app/controllers/application_controller.rb [hunk 2]
```diff
@@ -1,5 +1,10 @@
 class ApplicationController < ActionController::Base
-  def current_user
-    User.first
+  before_action :update_allowed_parameters, if: :devise_controller?
+
```
Comment (julie-ify) [inline app/controllers/application_controller.rb:4]: - [ ] Kindly add this helper method to ensure that users who are not logged in are always redirected to the login page, and cannot visit any other page.

  ```
  before_action :authenticate_user!
  ```

File: app/views/devise/registrations/new.html.erb [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    <%= f.label :password_confirmation %><br />
+    <%= f.password_field :password_confirmation, autocomplete: "new-password" %>
+  </div>
+
+  <div class="actions">
+    <%= f.submit "Sign up" %>
+  </div>
+<% end %>
+
+<%= render "devise/shared/links" %>
```
Comment (julie-ify) [inline app/views/devise/registrations/new.html.erb]: - [ ] Ensure your sign-up page matches the design. As you can see it is somewhat centered, but your own is on the left part of the page.

  kore

- [ ] Redesign your login page as well to match the design.

  kore

File: app/views/users/index.html.erb [hunk 4]
```diff
@@ -1,3 +1,4 @@
+<%= button_to "Sign out", destroy_user_session_path, :method => :delete %>
```
Comment (julie-ify) [inline app/views/users/index.html.erb:1]: - [ ] **[OPTIONAL]** If you look at the routes in the console you would see that the `destroy_user_session_path` route is `GET` request, not `DELETE`. I would recommend you fix it.

**Review by julie-ify — CHANGES_REQUESTED** [review]
## Hi @petushka1 and @c3c1l1a,

Good job so far!

## Highlights!
- Good feature branch name ✔️ 
- No linter errors ✔️ 
- Descriptive pull request ✔️ 
- Descriptive commit messages ✔️ 
- Good code organization and working app ✔️
- Descriptive README ✔️ 

Hello ✋, you have done well reaching this point of the milestone 🤝 . But there are some issues that you still need to work on to go to the next project but you are almost there!

# Status: Required Changes ♻️⛔️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread and tag me `@julie-ify` if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/views/devise/registrations/new.html.erb (hunk 3 above)
Comment (petushka1) [reply app/views/devise/registrations/new.html.erb]: can you please advise how I can generate only login view?

**Comment by petushka1** [comment]
# Debugging and improvement

- [x] add to routes `devise_for :users`
- [x] add rule to app controller `before_action :authenticate_user!`
- [x] integrate Bootstrap
- [x] regenerate views
- [x] style registration and sign in forms

**Review by DeliceLydia — APPROVED** [review]
## Hi @petushka1, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
# Congratulations! 🎉 



### Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**